Data/models/Bookmark.swift

```diff
@@ -561,17 +564,22 @@ extension Bookmark {
         // As for now, the return value for adding bookmark is never used.
     }
 
-    func updateResolvedRecord(_ record: SyncRecord?, context: WriteContext = .new(inMemory: false)) {
+    func updateResolvedRecord(_ record: SyncRecord?, context: WriteContext) {
         guard let bookmark = record as? SyncBookmark, let site = bookmark.site else { return }
-        title = site.title
-        syncParentUUID = bookmark.parentFolderObjectId
-        updateInternal(customTitle: site.customTitle, url: site.location,
-               newSyncOrder: bookmark.syncOrder, save: false, sendToSync: false, context: context)
-        lastVisited = Date(timeIntervalSince1970: (Double(site.lastAccessedTime ?? 0) / 1000.0))
-        if let recordCreated = record?.syncNativeTimestamp {
-            created = recordCreated
+        
+        DataController.perform(context: context) { context in
```

Comment:
Bug: this was calling some code on main thread, then created a new background thread, now all update code is under the same thread